var/spack/repos/builtin/packages/quip/package.py

```diff
@@ -0,0 +1,186 @@
+        # that Makefile.rules uses by default is not reliably available.
+        gitversion = self.version
+        makefile = FileFilter(join_path(self.build_directory, "Makefile.rules"))
+        makefile.filter(
+            r"^\s*GIT_VERSION\s*:=.*", f"GIT_VERSION := -D'GIT_VERSION=\"{gitversion}\"'"
+        )
+        #
+        # Makefile.config requires user input when writing Makefile.inc.
+        # Obviously in an automated environment like Spack we cannot
+        # rely on user input. Therefore we write Makefile.inc ourselves.
+        quip_arch = os.environ["QUIP_ARCH"]
+        make_dir = join_path(self.build_directory, "build", quip_arch)
+        make_inc = join_path(make_dir, "Makefile.inc")
+        os.makedirs(make_dir, exist_ok=True)
+        list_vars = [
+            "QUIP_ARCH",
+            "QUIP_ROOT",
+            "QUIP_INSTALLDIR",
+            "F77",
+            "F90",
+            "F95",
+            "CPLUSPLUS",
+            "MATH_LINKOPTS",
+            "EXTRA_LINKOPTS",
+            "HAVE_CP2K",
+            "HAVE_VASP",
+            "HAVE_TB",
+            "HAVE_PRECON",
+            "HAVE_LOFT",
+            "HAVE_ONIOM",
+            "HAVE_LOCAL_E_MIX",
+            "HAVE_QC",
+            "HAVE_GAP",
+            "HAVE_QR",
+            "HAVE_SCALAPACK",
+            "HAVE_FX",
+            "HAVE_SCME",
+            "HAVE_MTP",
+            "HAVE_MBD",
+            "HAVE_TTM_NF",
+            "HAVE_CH4",
+            "HAVE_NETCDF4",
+            "HAVE_MDCORE",
+            "HAVE_ASAP",
+            "HAVE_CGAL",
+            "HAVE_METIS",
+            "HAVE_LMTO_TBE",
+        ]
+        quip_root = os.environ["QUIP_ROOT"]
+        sizeof = join_path(quip_root, "bin", "find_sizeof_fortran_t")
+        with open(make_inc, "w") as fp:
+            fp.write(f"GIT_VERSION = \"{os.environ['GIT_VERSION']}\"\n")
+            for key in list_vars:
+                fp.write(f"{key} = {os.environ[key]}\n")
+        # Append SIZEOF_FORTRAN_T setting (without the next open statement
+        # the setting is prepended to Makefile.inc(?))
+        with open(make_inc, "a") as fp:
+            subprocess.run(["bash", sizeof], stdout=fp, check=True)
+
+    def build(self, spec, prefix):
+        #
+        # Note that os.chdir does not change PWD in the Python process
+        # so make will inherit the old PWD from Python. This causes
+        # problems when Makefile calls make for the fox library.
+        # To avoid that we need to update PWD explicitly. Just to make things
+        # more interesting os.getcwd does return the current working directory
+        # as it is after os.chdir.
+        os.chdir(self.build_directory)
+        build_path = os.getcwd()
+        os.environ["PWD"] = build_path
+        #
+        # With the correct environment we can call make.
+        make()
+
+    def install(self, spec, prefix):
+        make("install")
```

Comment:
This is should be unnecessary per https://spack.readthedocs.io/en/latest/build_systems/makefilepackage.html#phases.